rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java

```diff
@@ -0,0 +1,1105 @@
+
+    private enum FederationType {
+        erp,
+        p2sh,
+        standard
+    }
+
+    /***
+     * Key is BtcTxHash and output index. Value is the address that received the funds
+     * I can use this to validate that a certain redeemscript trying to spend this utxo generates the corresponding address
+     */
+    private Map<Sha256Hash, Address> whoCanSpendTheseUtxos = new HashMap<>();
+
+    private void testChangePowpeg(
+            FederationType federationTypeFrom,
+            List<Triple<BtcECKey, ECKey, ECKey>> keysFrom,
+            Address expectedAddressFrom,
+            List<UTXO> existingUtxos,
+            FederationType federationTypeTo,
+            List<Triple<BtcECKey, ECKey, ECKey>> keysTo,
+            Address expectedAddressTo,
+            BridgeConstants bridgeConstants,
+            ActivationConfig.ForBlock activations,
+            long migrationShouldFinishAfterThisAmountOfBlocks
+    ) throws Exception {
+
+        Repository repository = new MutableRepository(
+                new MutableTrieCache(
+                        new MutableTrieImpl(null, new Trie())
+                )
+        );
+        BridgeStorageProvider bridgeStorageProvider = new BridgeStorageProvider(
+                repository,
+                PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstants,
+                activations
+        );
+
+        BtcBlockStoreWithCache.Factory btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(
+                bridgeConstants.getBtcParams(),
+                100,
+                100
+        );
+
+        BtcBlockStore btcBlockStore = btcBlockStoreFactory.newInstance(repository, bridgeConstants, bridgeStorageProvider, activations);
+
+        // Setting a chain head different than genesis to avoid having to read the checkpoints file
+        addNewBtcBlockOnTipOfChain(btcBlockStore, bridgeConstants);
+
+        repository.save();
+
+        BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
+
+        long blockNumber = 0;
+
+        /***
+         * Creation phase
+         */
+
+        Block initialBlock = mock(Block.class);
+        when(initialBlock.getNumber()).thenReturn(blockNumber);
+
+        BridgeSupport bridgeSupport = new BridgeSupportBuilder()
+                .withProvider(bridgeStorageProvider)
+                .withRepository(repository)
+                .withEventLogger(bridgeEventLogger)
+                .withExecutionBlock(initialBlock)
+                .withActivations(activations)
+                .withBridgeConstants(bridgeConstants)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withPeginInstructionsProvider(new PeginInstructionsProvider())
+                .build();
+
+        List<FederationMember> originalPowpegMembers = keysFrom.stream().map(theseKeys -> new FederationMember(theseKeys.getLeft(), theseKeys.getMiddle(), theseKeys.getRight())).collect(Collectors.toList());
```

Comment:
Since you are not using the keys separately, just map each `Triple` to a `FederatinMember`, why not passing a list of `FederationMember` right away?

Reply:
Each Keys Is Actually Being Used Separately To Create Each FedMember via The getLeft/getMiddle/getRight Methods

Reply:
I want to leave the private keys to use this same test in the future to also test the signing of the pegouts